numpy/ma/core.py

```diff
@@ -4286,22 +4297,45 @@ def count(self, axis=None):
         array([3, 0])
 
         """
+        kwargs = {} if keepdims is np._NoValue else {'keepdims': keepdims}
+
         m = self._mask
-        s = self.shape
+        # special case for matrices (we assume no other subclasses modify
+        # their dimensions)
+        if isinstance(self.data, np.matrix):
+            if m is nomask:
+                m = np.zeros(self.shape, dtype=np.bool_)
+            m = m.view(type(self.data))
+
         if m is nomask:
-            if axis is None:
+            # compare to _count_reduce_items in _methods.py
+
+            if self.shape is ():
+                return 1
```

Comment:
Yeah, it is correct, though I wonder why it is needed (i.e. it could error on incorrect axis), but then, this probably just keeps old behaviour. (arr.ndim == 0; arr.count(axis=1) does not make much sense as such).